src/constraints/battery_degradation.jl

```diff
@@ -118,131 +118,62 @@ function add_degradation(m, p; b="ElectricStorage")
-            c[mth] = p.s.storage.attr[b].installed_cost_per_kwh * disc_factor
+            batt_replace_count = Int(ceil(N/day - 1)) # number of battery replacements in analysis period if they periodically happened on "day"
+            maint_cost = sum(p.s.storage.attr[b].degradation.maintenance_cost_per_kwh[day*i] for i in 1:batt_replace_count)
+            c[mth] = maint_cost
 
-            #= Salvage value logic and example
-            If replacement happens in month 145, then we will do 2 replacements, but only use last battery for ~6% of its expected life
-            In this case, salvage factor comes to 93.1%. This is the useful proportion of BESS life left after analysis period ends.
-            We account for this by creating a salvage value cost vector and subtracting it from the objective (i.e. maximizing this value).
-            
-            Logic:
-                Assuming the same replacement period as math, calculate the remaining fraction of useful life after analysis period
-                The BESS residual value is product of last day's maintenance_cost_per_kwh and salvage_factor
-                Set salvaged_value for that mth.
-            =#
-            salvage_factor = 1 - (p.s.financial.analysis_years*12/mth - floor(p.s.financial.analysis_years*12/mth))
-            salvaged_value = p.s.storage.attr[b].degradation.maintenance_cost_per_kwh[end]*salvage_factor
-            s[mth] = salvaged_value
+            residual_factor = 1 - (p.s.financial.analysis_years*12/mth - floor(p.s.financial.analysis_years*12/mth))
+            residual_value = p.s.storage.attr[b].degradation.maintenance_cost_per_kwh[end]*residual_factor
+            s[mth] = residual_value
             
         end
 
-        #=
-        -> linearize the product of bmth & m[:dvStorageEnergy][b]
-        For all months
-            0 <= bmth_BkWh <= dvStorageEnergy <= M (when bmth is true)
-
-            if bmth is 0, bmth_BkWh = 0. if bmth is 1, bmth_BkWh can equal m[:dvStorageEnergy][b]
-        =# 
-        M = p.s.storage.attr[b].max_kwh  # the big M
-        @variable(m, 0 <= bmth_BkWh[months])
-        @constraint(m, [mth in months], bmth_BkWh[mth] <= m[:dvStorageEnergy][b])
-        @constraint(m, [mth in months], bmth_BkWh[mth] <= M * bmth[mth])
-        @constraint(m, [mth in months], bmth_BkWh[mth] >= m[:dvStorageEnergy][b] - M*(1-bmth[mth]))
-
-        # add replacment cost to objective
+        # create replacement cost expression for objective
         @expression(m, degr_cost, sum(c[mth] * bmth_BkWh[mth] for mth in months))
 
-        # create salvage value expression for objective
-        @expression(m, salv_value, sum(s[mth] * bmth_BkWh[mth] for mth in months))
+        # create residual value expression for objective
+        @expression(m, residual_value, sum(s[mth] * bmth_BkWh[mth] for mth in months))
```

Comment:
is maintenance_cost_per_kwh actually a replacement cost?  we may want to rename it for clarity (so itn's not confused with O&M costs).